pages/api/work-dates/[nurseId].js

```diff
@@ -0,0 +1,44 @@
+import dbConnect from "../../../db/connect";
+import Nurse from "@/db/models/Nurse";
+import NurseWorkDates from "@/db/models/NurseWorkDates";
+
+export default async function handler(request, response) {
+  const connection = await dbConnect();
+  const { nurseId } = request.query;
+
+  if (!connection)
+    return response
+      .status(500)
+      .json({ error: "Unable to connect to the database!" });
+
+  if (request.method === "GET") {
+    try {
+      const nurse = await Nurse.findById(nurseId).populate("workSchedule");
+      if (!nurse) {
+        return response.status(404).json({ status: "Nurse not Found 🫣" });
+      }
+      response.status(200).json(nurse.workSchedule);
+    } catch (error) {
+      response.status(500).json({ status: "Error fetching nurse work dates." });
+    }
+  }
```

Comment:
please enter an "early return" inside each if statement. By that we can be sure that the function is terminated after one case is matched.